subprojects/jvm-services/src/test/groovy/org/gradle/jvm/toolchain/internal/IntellijIdeaInstallationSupplierTest.groovy

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.jvm.toolchain.internal
+
+import org.gradle.api.internal.provider.Providers
+import org.gradle.api.provider.ProviderFactory
+import org.gradle.internal.SystemProperties
+import org.gradle.internal.os.OperatingSystem
+import org.gradle.test.fixtures.file.TestNameTestDirectoryProvider
+import org.junit.Rule
+import spock.lang.Specification
+
+import static org.gradle.api.internal.file.TestFiles.systemSpecificAbsolutePath
+
+class IntellijIdeaInstallationSupplierTest extends Specification {
+
+    @Rule
+    public final TestNameTestDirectoryProvider temporaryFolder = new TestNameTestDirectoryProvider(getClass())
```

Comment:
Maybe add a test for windows.
Currently it seems that Windows / WSL install this in `$HOME/.jdks` as well : 

https://github.com/JetBrains/intellij-community/blob/c276879b0aa54d212acb0449f1866483471a7500/platform/lang-impl/src/com/intellij/openapi/projectRoots/impl/jdkDownloader/JdkInstaller.kt#L135-L141

Reply:
I added tests for Windows with 6f88697c8ecb3b6afda422a52758e3cff584caae. But I am not familiar enough with WSL to test and implement the behavior. Maybe some else can jump in?

Reply:
We can always improve the WSL aspect later, I wouldn't delay what we have for that reason. If somebody jumps in, fine, but we can merge in what we already have.